### ISSUE TITLE
Bug Report: roStoryReplace Unsupported function

### ISSUE THREAD

**Issue by PascalViauRC** [nrkno/sofie-mos-connection#86]
### About me

This issue is reported for CBC/Radio-Canada

### Observed Behavior

Using a connection with an iNews MOS Gateway, when I change the value of certain fields (eg. title, page-number), the iNews Gateway sends a message like this : 

```xml
{
    "mos": {
        "mosID": "SOFIE",
        "ncsID": "SRC",
        "roStoryReplace": {
            "roID": "SRC/PROD-MONTREAL._FORMATION.DEV-3.CONDUCTEUR",
            "storyID": 504309452,
            "story": {
                "storySlug": "01.01 - ----- on cam A sd<",
                "storyID": 504309452
            }
        }
    }
}
```

The roStoryReplace command does not update the story. I get this error in the logs : 
 
```
{"message":"Unsupported function","level":"debug"}
```

### Expected Behavior

The roStoryReplace command should update the story.

I've notice this code in the file MosDevices.ts (sofie-mos-connection/packages/connector/src/MosDevice.ts), line 429 : 

```js
	} else if (
		data.roElementAction?.operation === 'REPLACE' &&
		data.roElementAction.element_source?.story &&
		typeof this._callbackOnROReplaceStories === 'function'
	) {
		const action: IMOSStoryAction = {
			RunningOrderID: this.mosTypes.mosString128.create(data.roElementAction.roID),
			StoryID: this.mosTypes.mosString128.create(data.roElementAction.element_target?.storyID),
		}
		const sourceStories = data.roElementAction.element_source.story
		const stories: Array<IMOSROStory> = MosModel.XMLROStories.fromXML(
			Array.isArray(sourceStories) ? sourceStories : [sourceStories],
			this.strict
		)
		const resp = await this._callbackOnROReplaceStories(action, stories)
		return new MosModel.ROAck(resp, this.strict)
	} else if (
```
I tested adding this ```else if``` block just after, and it seems to do the trick : 

```js
	} else if (
		data.roStoryReplace && typeof this._callbackOnROReplaceStories === 'function'
	) {
		const action: IMOSStoryAction = {
			RunningOrderID: this.mosTypes.mosString128.create(data.roStoryReplace.roID),
			StoryID: this.mosTypes.mosString128.create(data.roStoryReplace.storyID),
		}
		const sourceStories = data.roStoryReplace.story
		const stories: Array<IMOSROStory> = MosModel.XMLROStories.fromXML(
			Array.isArray(sourceStories) ? sourceStories : [sourceStories],
			this.strict
		)
		const resp = await this._callbackOnROReplaceStories(action, stories)
		return new MosModel.ROAck(resp, this.strict)
	} else if (
```
I don't know if there is anything else to do though.

### Version

@mos-connection v3.0.6,  Sofie Core version 1.49.2

### Severity / Impact

We are in the process of testing Sofie. We are not currently in production.

**Comment by ianshade** [comment]
If I read the docs correctly, `roStoryReplace` is a MOS v2.6 message, that has been deprecated and replaced with `roElementAction` in v2.8. Unless there's a way to make the iNews MOS Gateway send `roElementAction`, I think support for messages 3.6.1. [roStoryAppend](https://mosprotocol.com/wp-content/MOS-Protocol-Documents/MOS_Protocol_Version_2.8.5_Final.htm#roStoryAppend) through 3.6.11. [roItemDelete](https://mosprotocol.com/wp-content/MOS-Protocol-Documents/MOS_Protocol_Version_2.8.5_Final.htm#roItemDelete) might need to be added to this library.

**Comment by PascalViauRC** [comment]
Thanks, let me check on our side if we can upgrade to MOS v2.8, so you won't have to support deprecated messages.

**Comment by nytamin** [comment]
I believe this is an issue where we misinterpreted the mos protocol when implementing the library.

While the `roStory***` and `roItem***` methods are to be deprecated in future version of MOS, it looks like they actually aren't deprecated in the current version.
Quote from the documentation:
> At present, applications and devices claiming support for MOS v2.8.5 Profile 2 must support all of these messages for backwards compatibility plus the functionally equivalent <[roElementAction](https://mosprotocol.com/wp-content/MOS-Protocol-Documents/MOS_Protocol_Version_2.8.5_Final.htm#roElementAction)> which will be forward compatible.

I've done an initial investigation and I believe it should be possible to add the missing methods as internal redirects to the existing functions. This means that the consumer-facing API will be unchanged, which is nice.

**Comment by PascalViauRC** [comment]
Thank would be great, thanks !

**Comment by PascalViauRC** [comment]
I just realize that this issue might be duplicate of that open issue : https://github.com/nrkno/sofie-mos-connection/issues/57

**Comment by nytamin** [comment]
I had some time to spare and made a possible fix for this in #87 .
@PascalViauRC would you be able to give that a spin and see if that helps with your case?
